### ISSUE TITLE
Nightmare returning wrong exit code

### ISSUE THREAD

**Issue by jpJuni0r** [hastern/nightmare#41]
The current implementation seems to check if the last test failed, otherwise it returns _0_ as an exit code.

However it should only return 0 when _all_ of the tests are okay, not just the last one.

https://github.com/hastern/nightmare/blob/5282a7573cbe2349efc0bb4c5b37da5789bf9c01/pyTestMain.py#L79

**Comment by hastern** [comment]
I've been thinking a bit on this. And even though I merged the pull request, I'm not sure the change is actually for the better.

Nightmare is intended to run until the first test fails. In case of a host application the exit code then is used to communicate the test result of the failed test. So either the host application needs to parse the program output (which can be silenced) or the I need to revert the changes.

**Comment by jpJuni0r** [comment]
I'm not sure if you fully understood the intention of my PR.

The current behavior without the patch is, when when creating a test suite with one failing and one passing tests the programs returns 0 as its exit code (see screenshot below).

At least for me this doesn't make sense as checking the exit code in CI environments or similar is the primary way to determine whether all tests ran successfully.

![2019-01-04-212507_634x282_scrot](https://user-images.githubusercontent.com/5278187/50709184-3944d900-1067-11e9-8d3c-7fad35dc6224.png)

**Comment by hastern** [comment]
I did understand that. I think I just stumbled on another bug. Originally nightmare was to stop after the first failed test, which is why the "--continue" option exists. But it seems to me, that nightmare currently always runs in continuous mode. So without the "--continue" option, your test bench should stop after the first test case (which they don't).

If the intention is to run all tests, than it obviously make no sense to convey information multiple tests using only a single number. If I remember correctly we did run each test individually in our CI, when I started the development some years ago. I have no idea if that has changed in the past.

I thinking of switching the behavior of the exit code depending on the TestSuiteMode, when running continuously the exit codes is the sum of all failed test otherwise its the old behavior. But to be honest I'm not sure if the is a good solutions. I'm open for suggestions.
